### PR TITLE
dev: buildkite separators for e2e server logs

### PR DESCRIPTION
**Pull request by keegancsmith**
Add in buildkite separators for the output of the server logs. This should help buildkite have different sections for each thing our e2e tests do. Note: this doesn't work just yet since we have vagrant including a prefix. But this is a step in the right direction.

### CODE REVIEW

File: dev/ci/e2e.sh [hunk 1]
```diff
@@ -18,7 +18,15 @@ fi
...
 echo "--- Running a daemonized $IMAGE as the test subject..."
 CONTAINER="$(docker container run -d "$IMAGE")"
-trap 'kill $(jobs -p -r)'" ; docker logs --timestamps $CONTAINER ; docker container rm -f $CONTAINER ; docker image rm -f $IMAGE" EXIT
+function cleanup() {
+  jobs -p -r | xargs kill
+  echo "--- server logs"
+  docker logs --timestamps "$CONTAINER"
+  echo "--- docker cleanup"
+  docker container rm -f "$CONTAINER"
+  docker image rm -f "$IMAGE"
```
Comment (mrnugget) [inline dev/ci/e2e.sh:27]: Do we even need to log this?

**Review by mrnugget — APPROVED** [review]

File: dev/ci/e2e.sh (hunk 1 above)
Comment (keegancsmith) [reply dev/ci/e2e.sh:27]: useful incase it fails.

File: dev/ci/e2e.sh [hunk 2]
```diff
@@ -45,6 +53,7 @@ echo "Waiting for $URL... done"
 echo "--- yarn run test-e2e"
 env SOURCEGRAPH_BASE_URL="$URL" PERCY_ON=true ./node_modules/.bin/percy exec -- yarn run cover-e2e
 
+echo "--- coverage"
 yarn nyc report -r json
```
Comment (felixfbecker) [inline dev/ci/e2e.sh:57]: I believe this is not what prints the coverage to the terminal, this is what outputs the JSON report to a file (`--reporter json`). The coverage to the terminal I believe is printed by the `yarn run cover-e2e` command above, which runs `nyc`, which defaults to outputting the report at the end. I think that's where we need to add a command line flag to supress that, or set it through `nyc` config in package.json (should be easy to do).
If we want to keep the output we can add an explicit `nyc report` call which we can nest under a Buildkite section (but I don't need it).

Comment (keegancsmith) [reply dev/ci/e2e.sh:57]: yeah, I added it as separate so it wouldn't be included in the above commands output.

**Pull request merged**